### PR TITLE
quoting default pathToJest

### PR DESCRIPTION
**Pull request by connectdotz**
this is to follow up jest-community/jest-editor-support#21:

with `jest-editor-support` 27.0.0 now able to pass raw pathToJest to the spawn call, users can embed escape characters or quotes when they customize `jest.pathToJest` and our generated default jestToPath should do the same in case there is escape sequence in user's path. 

related issues:
#342
#360
#426
#489

### CODE REVIEW

**Comment by connectdotz** [comment]
@omjadas @stephtr @seanpoulter does any of you have a few minutes to review so we can close this issue and cut a new release soon...

File: src/helpers.ts [hunk 1]
```diff
@@ -79,7 +79,8 @@ export function pathToJest({ pathToJest, rootPath }: IPluginResourceSettings) {
     return 'npm test --'
   }
 
-  return getLocalPathForExecutable(rootPath, 'jest') || 'jest' + nodeBinExtension
+  const p = getLocalPathForExecutable(rootPath, 'jest') || 'jest' + nodeBinExtension
```
Comment (stephtr) [inline src/helpers.ts]: I don't have it in mind, if we always use the shell, do we need `nodeBinExtension` on Windows?

**Comment by stephtr** [comment]
Beside the shell option it looks good to me.

File: src/helpers.ts (hunk 1 above)
Comment (connectdotz) [reply src/helpers.ts]: I don't have a windows machine to verify, if it worked without, I can remove it... does it matter which shell is used? I could imagine users might want to pass their favor shell in the future...

**Comment by connectdotz** [comment]
@omjadas thanks for reminding me about the deprecated shell option, it's now fixed.

@stephtr I am glad you are back! I agree we should remove the `nodeBinExtension` for windows if we could, but would definitely need to test it before committing. Since I didn't have a way to verify this, I think it's best to leave it as it is until maybe you or someone else to submit a separate PR for windows.

**Comment by connectdotz** [comment]
@stephtr @omjadas is there any other change that should be considered? If not please approve otherwise Github blocks merging.

**Comment by stephtr** [comment]
> I agree we should remove the nodeBinExtension for windows if we could, but would definitely need to test it before committing. Since I didn't have a way to verify this, I think it's best to leave it as it is until maybe you or someone else to submit a separate PR for windows.

That's true, especially since we already had a few issues related to that.

**Pull request merged**